XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs

```diff
@@ -204,96 +203,78 @@ protected override void PageLeaveCore(PageLoadedDirection direction, ref bool ca
 
-                    if (downloadUpdateRadioButton.Checked)
-                    {
-                        var distinctHosts = SelectedUpdateAlert != null ? SelectedUpdateAlert.DistinctHosts : null;
-                        if (distinctHosts != null && distinctHosts.Any(Helpers.ElyOrGreater))
-                            // this is to check whether the Alert represents an ISO update (Ely or greater)
-                        {
-                            SelectedUpdateType = UpdateType.ISO;
-                        }
-                        else //legacy format
-                        {
-                            SelectedUpdateType = UpdateType.Legacy;
-                        }
-                    }
-                    else
-                    {
-                        if (WizardHelpers.IsValidFile(fileName))
+                    FileFromDiskAlert = null;
+                    FileFromDiskHasUpdateXml = false;
+                    unzippedUpdateFilePath = null;
+                    SelectedNewPatch = null;
+                }
+                else
+                {
+                    SelectedUpdateAlert = null;
+                    SelectedNewPatch = null;
+
+                    if (!WizardHelpers.IsValidFile(FilePath, out var pathFailure))
+                        using (var dlg = new ThreeButtonDialog(new ThreeButtonDialog.Details(
+                            SystemIcons.Error, pathFailure, Messages.UPDATES)))
                         {
-                            if (fileName.EndsWith("." + Branding.Update))
-                                SelectedUpdateType = UpdateType.Legacy;
-                            else if (fileName.EndsWith("." + Branding.UpdateIso))
-                                SelectedUpdateType = UpdateType.ISO;
-                            else
-                                SelectedUpdateType = UpdateType.Existing;
+                            cancel = true;
+                            dlg.ShowDialog();
+                            return;
                         }
-                    }
 
-                    if (SelectedExistingPatch != null && !SelectedExistingPatch.Connection.IsConnected)
-                    {
-                        cancel = true;
+                    SelectedNewPatch = FilePath;
 
-                        using (var dlg = new ThreeButtonDialog(
-                            new ThreeButtonDialog.Details(SystemIcons.Warning,
-                                string.Format(Messages.UPDATES_WIZARD_CANNOT_DOWNLOAD_PATCH, SelectedExistingPatch.Connection.Name),
-                                Messages.UPDATES_WIZARD)))
+                    if (!string.IsNullOrEmpty(FilePath) && Path.GetExtension(FilePath).ToLowerInvariant().Equals(".zip"))
+                    {
+                        //check if we are installing update user sees in textbox
+                        if (unzippedUpdateFilePath == null || Path.GetFileNameWithoutExtension(unzippedUpdateFilePath) !=

```

Comment:
If the unzipped file is not there anymore (for example because I cleaned my Temp folder), the wizard shows the invalid file errror, instead of trying to unzip it again.